cpp/src/groupby/groupby.cu

```diff
@@ -211,5 +213,25 @@ detail::sort::sort_groupby_helper& groupby::helper()
   return *_helper;
 };
 
+std::pair<std::unique_ptr<table>, std::unique_ptr<column>> groupby::shift(
+  column_view const& values,
+  size_type offset,
+  scalar const& fill_value,
+  rmm::mr::device_memory_resource* mr)
+{
+  CUDF_FUNC_RANGE();
+  CUDF_EXPECTS(values.type() == fill_value.type(),
+               "values and fill_value should have the same type.");
+
+  auto stream        = rmm::cuda_stream_default;
+  auto sorted_keys   = helper().sorted_keys(stream);
+  auto sorted_values = helper().sorted_values(values, stream, mr);
```

Comment:
The values within group should not be sorted for shift.
Compared the expected behavior with pandas or Spark.

Also, since this is temporary data, `mr` shouldn't be passed.
```suggestion
  auto grouped_values = helper().grouped_values(values, stream);
```